model/user/user.go

```diff
@@ -172,7 +173,36 @@ func (u *DBUser) IncPatchNumber() (int, error) {
 		return 0, err
 	}
 	return dbUser.PatchNumber, nil
+}
+
+func (u *DBUser) PatchFinishSubscriber() (*event.Subscriber, error) {
+	if !IsValidSubscriptionPreference(string(u.Settings.Notifications.PatchFinish)) {
+		return nil, errors.Errorf("unknown subscriber preference: %s", u.Settings.Notifications.PatchFinish)
+	}
+
+	return u.subscriber(u.Settings.Notifications.PatchFinish), nil
+}
+
+func (u *DBUser) BuildBreakSubscriber() (*event.Subscriber, error) {
```

Comment:
it looks like we'll need to modify the implementation of users for every new subscription type. Is there a way that we can refactor this to avoid (or minimize/control) the extent to which a future developer would need to edit the codebase in multiple places (triggers, notifications, users, events, units(?), I may be incorrect) to add a new kind of event while *also* avoiding needing to push notification related functionality into the top-level user interface.